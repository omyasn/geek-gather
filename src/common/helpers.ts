export const formatToHumanDate = (date: string) => (
    new Date(date).toLocaleDateString(
        'en-gb',
        {
          year: 'numeric',
          month: 'long',
          day: 'numeric'
        }
      )
);