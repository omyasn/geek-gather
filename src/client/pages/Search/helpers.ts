import { EventType } from '../../../common/commonTypes';

export const intersection= (arrays: Array<string>[]): string[] => {
    if (arrays.length === 0) { return []; }
    if (arrays.length === 1) { return arrays[0]; }

    let acc = arrays[0];
    for (let i = 1; i < arrays.length; i++) {
        acc = acc.filter(el => arrays[i].includes(el));
        if (acc.length === 0) { return []; }
    }
    return acc;

    //return arrays.reduce((acc, cur) => acc.filter(i => cur.includes(i)))
};

export const mapOfEvents = (events: EventType[]): Map<string, EventType> => {
    const eventsMap = new Map();
    events.forEach(event => eventsMap.set(event.id, event));

    return eventsMap;
};

export function createOrAddToSet<T>(container: Set<T>, value: T) {
    return container ? container.add(value) : new Set([value]);
}