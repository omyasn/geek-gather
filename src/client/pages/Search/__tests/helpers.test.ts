import { intersection, mapOfEvents, createOrAddToSet } from '../helpers';

test('intersection', () => {
    const test = intersection([['1', '2', '4'], ['2', '3', '4'], ['4', '4', '2']]);
    expect(test).toEqual(['2', '4']);
});

test('intersection only one', () => {
    const test = intersection([['1', '2', '4']]);
    expect(test).toEqual(['1', '2', '4']);
});

test('intersection empty', () => {
    const test = intersection([]);
    expect(test).toEqual([]);
});


test('intersection no inter', () => {
    const test = intersection([['1', '2', '4'], ['0', '3', '9'], ['4', '5', '6']]);
    expect(test).toEqual([]);
});


test('mapOfEvents', () => {
    const test = mapOfEvents([
        {
            id: "evnt9",
            title: "Big Bears War",
            beginDate: "2022-08-07",
            capacity: 400,
            parties: "2",
            minPrice: 1200,
            private: false,
            hasQuests: true,
            ownBalls: true,
            location: "Raining Forest Club",
            city: "St.Peterbugr",
            locationLink: "https://google.com",
            owner: "It Is What It Is",
            ownerLink: "https://google.com",
          },
          {
            id: "evnt10",
            title: "Game for all",
            beginDate: "2022-08-10",
            location: "Centrum Club",
            city: "Noginsk",
            locationLink: "https://google.com",
            owner: "Peter",
            ownerLink: "https://google.com",
            capacity: 30,
            parties: "2",
            minPrice: 550,
            private: true,
            hasQuests: false,
            ownBalls: false,
          },
    ]);
    expect(test).toMatchSnapshot();
});


describe('createOrAddToSet', () => {
    it('create set', () => {
        const result: any = {};
        const value = 'val';
        result.attr = createOrAddToSet(result.attr, value);

        expect(result).toEqual({ attr: new Set(['val']) });
    });

    it('add to set', () => {
        const result: any = { attr: new Set(['lol']) };
        const value = 'kek';
        result.attr = createOrAddToSet(result.attr, value);

        expect(result).toEqual({ attr: new Set(['lol', 'kek']) });
    });
});