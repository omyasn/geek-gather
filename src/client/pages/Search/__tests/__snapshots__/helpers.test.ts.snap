// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapOfEvents 1`] = `
Map {
  "evnt9" => Object {
    "beginDate": "2022-08-07",
    "capacity": 400,
    "city": "St.Peterbugr",
    "hasQuests": true,
    "id": "evnt9",
    "location": "Raining Forest Club",
    "locationLink": "https://google.com",
    "minPrice": 1200,
    "ownBalls": true,
    "owner": "It Is What It Is",
    "ownerLink": "https://google.com",
    "parties": "2",
    "private": false,
    "title": "Big Bears War",
  },
  "evnt10" => Object {
    "beginDate": "2022-08-10",
    "capacity": 30,
    "city": "Noginsk",
    "hasQuests": false,
    "id": "evnt10",
    "location": "Centrum Club",
    "locationLink": "https://google.com",
    "minPrice": 550,
    "ownBalls": false,
    "owner": "Peter",
    "ownerLink": "https://google.com",
    "parties": "2",
    "private": true,
    "title": "Game for all",
  },
}
`;
